TrappingRainWater/main_test.cpp

```diff
@@ -0,0 +1,53 @@
+#include "main.h"
+
+#include <gtest/gtest.h>
+
+class SolutionTest : public ::testing::Test {
+ protected:
+  Solution solution;
+};
+
+TEST_F(SolutionTest, Example1) {
+  std::vector<int> height = {0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1};
+  EXPECT_EQ(solution.trap(height), 6);
+}
+
+TEST_F(SolutionTest, Example2) {
+  std::vector<int> height = {4, 2, 0, 3, 2, 5};
+  EXPECT_EQ(solution.trap(height), 9);
```

Comment:
you can write it like this: 
EXPECT_EQ(solution.trap({4, 2, 0, 3, 2, 5}), 9);

and then just put all the examples into one test.